### PR TITLE
Fix various multiplayer test failures due to not waiting for ongoing operation

### PR DESCRIPTION
**Pull request by peppy**
As seen at
https://teamcity.ppy.sh/buildConfiguration/Osu_Build/1138?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandBuildTestsSection=true.

### CODE REVIEW

File: osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayerMatchSongSelect.cs [hunk 1]
```diff
@@ -110,6 +110,7 @@ public void TestBeatmapConfirmed()
             AddUntilStep("wait for selection", () => Beatmap.Value.BeatmapInfo.Equals(selectedBeatmap));
             AddStep("set mods", () => SelectedMods.Value = new[] { new TaikoModDoubleTime() });
 
+            AddUntilStep("wait for ongoing operation to complete", () => !OnlinePlayDependencies.OngoingOperationTracker.InProgress.Value);
```
Comment (smoogipoo) [inline osu.Game.Tests/Visual/Multiplayer/TestSceneMultiplayerMatchSongSelect.cs:113]: Move this above the `set mods` step.

File: osu.Game.Tests/Visual/Multiplayer/TestSceneAllPlayersQueueMode.cs [hunk 2]
```diff
@@ -146,6 +147,7 @@ private void addItem(Func<BeatmapInfo> beatmap, RulesetInfo? ruleset = null, IRe
             if (mods != null)
                 AddStep($"set mods to {string.Join(",", mods.Select(m => m.Acronym))}", () => songSelect.AsNonNull().Mods.Value = mods);
 
+            AddUntilStep("wait for ongoing operation to complete", () => !(CurrentScreen as OnlinePlayScreen).ChildrenOfType<OngoingOperationTracker>().Single().InProgress.Value);
```
Comment (smoogipoo) [inline osu.Game.Tests/Visual/Multiplayer/TestSceneAllPlayersQueueMode.cs:150]: Move this above the `set {ruleset.Name} ruleset` step.

**Review by smoogipoo — CHANGES_REQUESTED** [review]
If this ever changes in the future so that mods / ruleset can't be altered during the operation (should probably be done), the tests would break silently again.

File: osu.Game.Tests/Visual/Multiplayer/TestSceneAllPlayersQueueMode.cs (hunk 2 above)
Comment (peppy) [reply osu.Game.Tests/Visual/Multiplayer/TestSceneAllPlayersQueueMode.cs:150]: Should there be one for each, or is checking only once enough? It's basically the room join operation, right?

Comment (smoogipoo) [reply osu.Game.Tests/Visual/Multiplayer/TestSceneAllPlayersQueueMode.cs:150]: Checking only once is enough. Not entirely sure exactly which things are considered operations, joining room could be done, changing items is definitely one.

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**